internal/core/queries/get_can_bus_dump_files_by_eth_address.go

```diff
@@ -0,0 +1,58 @@
+package queries
+
+import (
+	"context"
+	"fmt"
+	"github.com/DIMO-Network/vehicle-signal-decoding/internal/config"
+	"github.com/DIMO-Network/vehicle-signal-decoding/internal/infrastructure/exceptions"
+	p_grpc "github.com/DIMO-Network/vehicle-signal-decoding/pkg/grpc"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/rs/zerolog"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type GetCanBusDumpFileByEthAddressQueryHandler struct {
+	logger   *zerolog.Logger
+	s3Client *s3.Client
+	settings *config.Settings
+}
+
+func NewGetCanBusDumpFileByEthAddressQueryHandler(logger *zerolog.Logger, s3Client *s3.Client, settings *config.Settings) GetCanBusDumpFileByEthAddressQueryHandler {
+	return GetCanBusDumpFileByEthAddressQueryHandler{
+		logger:   logger,
+		s3Client: s3Client,
+		settings: settings,
+	}
+}
+
+type GetCanBusDumpFileByEthAddressQueryRequest struct {
+	EthAddress string
+}
+
+func (h GetCanBusDumpFileByEthAddressQueryHandler) Handle(ctx context.Context, query *GetCanBusDumpFileByEthAddressQueryRequest) (*p_grpc.GetCanBusDumpFileResponse, error) {
+
+	response, err := h.s3Client.ListObjectsV2(ctx, &s3.ListObjectsV2Input{
+		Bucket: aws.String(h.settings.AWSDocumentsBucketName),
+		Prefix: aws.String(query.EthAddress + "/"),
```

Comment:
we will need to make sure EthAddress string representation starts with `0x`, maybe we can add validation somewhere.